test/AspNet.Security.OAuth.Providers.Tests/Xumm/XummTests.cs

```diff
@@ -0,0 +1,82 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using Microsoft.AspNetCore.WebUtilities;
+
+namespace AspNet.Security.OAuth.Xumm;
+
+public class XummTests : OAuthTests<XummAuthenticationOptions>
+{
+    public XummTests(ITestOutputHelper outputHelper)
+    {
+        OutputHelper = outputHelper;
+    }
+
+    public override string DefaultScheme => XummAuthenticationDefaults.AuthenticationScheme;
+
+    protected internal override void RegisterAuthentication(AuthenticationBuilder builder)
+    {
+        builder.AddXumm(options => ConfigureDefaults(builder, options));
+    }
+
+    [Theory]
+    [InlineData(ClaimTypes.NameIdentifier, "my-r-address")]
+    [InlineData(ClaimTypes.Name, "John Smith")]
+    [InlineData(ClaimTypes.Email, "john.smith@xumm.me")]
+    [InlineData(XummAuthenticationConstants.Claims.Picture, "Avatar")]
+    public async Task Can_Sign_In_Using_Xumm(string claimType, string claimValue)
+    {
+        // Arrange
+        using var server = CreateTestServer();
+
+        // Act
+        var claims = await AuthenticateUserAsync(server);
+
+        // Assert
+        AssertClaim(claims, claimType, claimValue);
+    }
+
+    [Theory]
+    [InlineData(false)]
+    [InlineData(true)]
+    public async Task BuildChallengeUrl_Generates_Correct_Url(bool usePkce)
```

Comment:
This test can be removed as your handler doesn't override `ExchangeCodeAsync()`, so it's just testing the base class.

Reply:
Updated to the recommended version.

Reply:
You're right, the redundant test has been removed.